clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -381,18 +377,23 @@ public boolean matches(ClientRequest request) {
             }
         }, new HeartbeatResponse(Errors.NONE.code()).toStruct(), coordinator);
 
+        // heartbeat interval is 2 seconds
+        time.sleep(heartbeatIntervalMs);
+        Thread.sleep(heartbeatIntervalMs);
```

Comment:
Hmm, this is kind of unfortunate. I assume we started having to do this since we have a separate thread that uses `wait` to handle time, so mock time doesn't work for it?